pkg/sqlcmd/sqlcmd.go

```diff
@@ -133,7 +135,11 @@ func (s *Sqlcmd) Run(once bool, processAll bool) error {
 				args = make([]string, 0)
 				once = true
 			} else {
-				_, _ = s.GetOutput().Write([]byte(err.Error() + SqlcmdEol))
+				if iactive {
+					_, _ = s.GetOutput().Write([]byte(err.Error() + SqlcmdEol))
+				} else {
+					_, _ = s.GetError().Write([]byte(err.Error() + SqlcmdEol))
```

Comment:
Don't you need to distinguish the case of iactive being true because of the user name parameter being passed in, which is part of your other PR?

Reply:
maybe Sqlcmd should have a property like "WriteErrorsToErrorStream bool" that would be set by main before it calls Run.

There are some other settings that control command output destination that this could be like.

Reply:
Added flag in sqlcmd struct to indicate an interactive session.
The interactive session will be determined when the first input is scanned.

Reply:
would it make sense to fix s.GetError to do the right thing based on context and just change this code to use GetError() only? 
I'm not 100% certain that every kind of error in sqlcmd goes to the same stream, though.